horovod/torch/mpi_ops.py

```
@@ -109,9 +109,13 @@ def _allreduce_async(tensor, output, name, op):
                     raise NotImplementedError('Running GPU Adasum with non-power of 2 nodes is not supported yet.')
                 divisor = local_size()
             else:
-                warnings.warn('Adasum reduction does not currently support GPU reduction using MPI. Tensors are '
-                              'copied to CPU memory instead. To use Adasum for GPU reduction, please compile Horovod '
-                              'with HOROVOD_GPU_ALLREDUCE=NCCL.')
+                if local_size() != 8:
```

So, this means that the code only works DGX1. We should change the logic such that it check for DGX1 connection, otherwise we should throw an error. As in cases where there are 8 GPUs in a node but not connected like a DGX1.